docs/services/ml/upload-training-script.md

```diff
@@ -0,0 +1,387 @@
+            data_augmentation,
+            base_model,
+            global_pooling,
+            classification,
+        ]
+    )(x)
+
+    model = tf.keras.Model(x, y)
+
+    model.compile(
+        loss=loss_fnc,
+        optimizer=tf.keras.optimizers.Adam(learning_rate=1e-3),
+        metrics=[metrics],
+    )
+    return model
+```
+
+{{% alert title="Tip" color="tip" %}}
+You must log in to the [Viam app](https://app.viam.com/) to download the package containing the classification training script from the registry.
+{{% /alert %}}
+
+{{% /expand%}}
+{{%expand "Step 4: Save the model artifact the script produces" %}}
+
+You also need to include logic in your <file>training.py</file> to save the model artifact your training job produces to the `model_output_directory`.
+For example:
+
+```python
+# Save the model artifact to the Viam registry using the provided
+# ML model name and version
+def save_tflite_classification(
+    model: Model,
+    model_dir: str,
+    model_name: str,
+    target_shape: ty.Tuple[int, int, int],
+) -> None:
+    # Convert the model to tflite, with batch size 1
+    # so the graph does not have dynamic-sized tensors.
+    input = tf.keras.Input(target_shape, batch_size=1, dtype=tf.uint8)
+    output = model(input, training=False)
+    wrapped_model = tf.keras.Model(inputs=input, outputs=output)
+    converter = tf.lite.TFLiteConverter.from_keras_model(wrapped_model)
+    converter.target_spec.supported_ops = TFLITE_OPS
+    tflite_model = converter.convert()
+
+    filename = os.path.join(model_dir, f"{model_name}.tflite")
+    with open(filename, "wb") as f:
+        f.write(tflite_model)
+```
+
+When you submit a training job with this training script, this function saves the model outputs to the `model_output_directory` in the cloud.
+Once the training job is complete, Viam looks at that directory and creates a package with all of the contents of the directory, creating or updating a registry item for the ML model.
+
+{{% /expand%}}
+
+### Package the training script as a <file>tar.gz</file> source distribution
+
+You must save your training script in the `tar.gz` format to run in the Viam ML training service.
+Follow the instructions to [create a <file>tar.gz</file> gzip'd tar file](https://docs.python.org/3.10/distutils/sourcedist) from your project.
+
+{{% alert title="Tip" color="tip" %}}
+You can reference the directory structure of this [example classification training script](https://app.viam.com/packages/e76d1b3b-0468-4efd-bb7f-fb1d2b352fcb/custom-training-classification/ml_training/latest/e76d1b3b-0468-4efd-bb7f-fb1d2b352fcb).
+You must log in to the Viam app to download the package.
+Unzip the package and see <file>model/training.py</file> for an example entrypoint file.
+{{% /alert %}}
+
+## Upload a new training script or new version
+
+You must use the [Viam CLI](/cli/) to upload your training script to the Registry.
+Follow the instructions in the [CLI documentation](/cli/) to install the CLI.
```

Comment:
```suggestion
```
If you are covering installing you'd then also need to cover authentication. I think this is such an advanced use case, we can assume they likely have the cli

Reply:
cool